lib/src/components/NxCombobox/__tests__/NxCombobox.test.tsx

```diff
@@ -0,0 +1,427 @@
+  });
+
+  it('sets aria-live on the alert dropdown to "polite"', function() {
+    expect(rtlRender().container.querySelector('.nx-combobox__alert')).toHaveAttribute('aria-live', 'polite');
+  });
+
+  it('sets aria-busy on the alert dropdown if loading is true', function() {
+    expect(rtlRender().container.querySelector('.nx-combobox__alert')).toHaveAttribute('aria-busy', 'false');
+    expect(rtlRender({ loading: true }).container.querySelector('.nx-combobox__alert'))
+        .toHaveAttribute('aria-busy', 'true');
+  });
+
+  it('renders a loading spinner when the `loading` prop is true', function() {
+    const { container } = rtlRender({ loading: true }),
+        loadingElement = container.querySelector('.nx-loading-spinner');
+    expect(loadingElement).toBeTruthy();
+  });
+
+  it('renders error when the `loadError` prop is set', function() {
+    const { container } = rtlRender({ loadError: 'err' }),
+        errorElement = container.querySelector('.nx-alert--load-error');
+    expect(errorElement).toBeTruthy();
+  });
+
+  it('fires onSearch with the search text when the retry button is clicked', async function() {
+    const onSearch = jest.fn(),
+        { container } = rtlRender({ loadError: 'err', onSearch }),
+        retryBtn = container.querySelector('.nx-load-error__retry')!;
+
+    expect(onSearch).not.toHaveBeenCalled();
+    await userEvent.click(retryBtn);
+    expect(onSearch).toHaveBeenCalled();
+  });
+
+  it('fires onChange when the button with role option is clicked', async function() {
+    const onChange = jest.fn(),
+        { getByRole, getAllByRole } = rtlRender({
+          matches: [{ id: '1', displayName: 'Foo' }, { id: '2', displayName: 'Boo' }], onChange }),
+        inputElement = getByRole('combobox');
+
+    await userEvent.type(inputElement, 'o');
+    const optionBtns = getAllByRole('option');
+    await userEvent.click(optionBtns[1]);
+    expect(onChange).toHaveBeenCalledWith('Boo');
+  });
+
+  it('renders empty message if there are no results', function() {
+    const { container } = rtlRender(),
+        emptyMessage = container.querySelector('.nx-combobox__empty-message');
+
+    expect(emptyMessage).toBeTruthy();
+    expect(emptyMessage).toHaveTextContent('No Results Found');
+  });
+
+  it('sets the empty message from the `emptyMessage` prop', function() {
+    const { container } = rtlRender({ emptyMessage: 'asdfasdf' }),
+        emptyMessage = container.querySelector('.nx-combobox__empty-message');
+
+    expect(emptyMessage).toHaveTextContent('asdfasdf');
+  });
+
+  it('calls onSearch with the current trimmed search text if focus enters the component from elsewhere on the page ' +
+  'while there is an error', function() {
+    const onSearch = jest.fn(),
+        { getByRole} = rtlRender({ value: 'f', loadError: 'err', onSearch }),
+        inputElement = getByRole('combobox'),
+        anotherElement = document.createElement('button');
+
+    document.body.append(anotherElement);
```

Comment:
This doesn't get cleaned up after the test.  A better way would be to use RTL's `render()` function, passing it a fragment that contains both the button and the combobox.

Reply:
Thanks for suggestion, updated here 4029d58f1778f46cf53cdedad783b595e4a9da4e.